ethcore/src/blockchain/blockchain.rs

```diff
@@ -1501,18 +1500,17 @@ mod tests {
 	#[test]
 	fn should_cache_best_block() {
 		// given
-		let mut canon_chain = ChainGenerator::default();
-		let mut finalizer = BlockFinalizer::default();
-		let genesis = canon_chain.generate(&mut finalizer).unwrap();
-		let first = canon_chain.generate(&mut finalizer).unwrap();
+		let genesis = BlockBuilder::genesis();
+		let first = genesis.add_block();
+		let mut generator = BlockGenerator::new(vec![genesis, first]);
```

Comment:
IMHO `BlockGenerator` should not be `Iterator`, but rather `IntoIterator` (Maybe even call it `BlockChain`)
```
let generator = BlockGenerator::new(&[genesis, first]);
new_chain(&generator[0].data())
// Retrieve iterator
let mut it = generator.iter();

// or consume generator and do sth with blocks
for b in generator {
  ...
}